src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerServiceValidateTest.java

```diff
@@ -40,18 +43,30 @@ class GeneralPartnerServiceValidateTest {
     private static final String GENERAL_PARTNER_ID = GeneralPartnerBuilder.GENERAL_PARTNER_ID;
     private static final String TRANSACTION_ID = TransactionBuilder.TRANSACTION_ID;
 
-    private final Transaction transaction = new TransactionBuilder().forPartner(
-            FILING_KIND_GENERAL_PARTNER,
-            URL_GET_GENERAL_PARTNER,
-            GENERAL_PARTNER_ID
-    ).build();
+    private Transaction transaction;
 
     @Autowired
     private GeneralPartnerService service;
 
     @MockitoBean
     private GeneralPartnerRepository repository;
 
+    @MockitoBean
+    private CompanyService companyService;
+
+    @MockitoBean
+    private CompanyProfileApi companyProfileApi;
+
+    @BeforeEach
+    void setup() throws ServiceException {
+        transaction = new TransactionBuilder().forPartner(
```

Comment:
moved transaction into beforeEach as one of the new tests will set the filingMode to transition and we don't want that set for other tests, so reset it each time here.